### DISCUSSION TITLE
Adding / Updating Notebooks. Separate repository with automatic sync to ipfs through github actions?

### DISCUSSION THREAD

**Discussion by voodoohop** [pollinations/pollinations#90] [Pollinations]
Adding / Updating Notebooks. 

Separate repository with automatic sync to ipfs through github actions.

I have gone the multi-repo path in the past and regretted it sometimes. But I think it makes sense in this case. Especially because our notebooks on git allow easy editing and saving from colab.

On each commit we would trigger the github ipfs action https://github.com/aquiladev/ipfs-action/ and would then update an ipns record with the new hash pointing to the root of the notebooks.

It's also completely unnecessary to redeploy the whole site just when we update a notebook.  this would be alleviated

**Comment by voodoohop** [comment]
@gokaykucuk any thoughts on this?

**Comment by gokaykucuk** [comment]
@voodoohop I am 100% on this. Imo there are 4 concerns right now on this repo,
- Pollinations.ai frontend
- Pollinator
- IPFS Node
- Generative notebooks

Sorry I'm on my phone now but I can write more details on the subject later on. But for start, extracting the notebooks and setting up github actions to deploy them seems the most urgent.

**Reply by voodoohop** [reply]
Should we move to this "@" mode people use to group projects on github? i dont completely get the ins and outs of it.

I see a lot of people moving back to the "monorepo" mode and myself was also bitten by having to keep different github repos that depend on each other in sync. So maybe I wouldn't  move to splitting everything into separate repositories just yet.

The generative notebooks for sure though. The others may make sense too. Let's think about it.

**Comment by voodoohop** [comment]
First step: https://github.com/pollinations/pollinations/issues/91
